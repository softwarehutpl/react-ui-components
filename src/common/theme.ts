const theme: { [key: string]: { [key: string]: { [key: string]: string } } } = {
  colors: {
    primary: {
      base: '#000000',
      light: '#ffffff',
      dark: '#a8b3bd',
    },
    secondary: {
      base: '#3690e3',
      light: '#75b0e6',
      dark: '#034d91',
    },
    error: {
      base: '#f44336',
      light: '#f56c62',
      dark: '#a1140a',
    },
    warning: {
      base: '#faa72d',
      light: '#ffc36b',
      dark: '#995b00',
    },
    info: {
      base: '#7979d4',
      light: '#9797d1',
      dark: '#4c4cc7',
    },
    success: {
      base: '#4caf50',
      light: '#88c28a',
      dark: '#124f14',
    },
    disabled: {
      base: '#d9d9d9',
      light: '#f0eded',
      dark: '#8c8989',
    },
  },
  // TODO: change it in dedicated task for that (change ease-out for example and add more options)
  transitions: {
    all: {
      fast: 'all 0.5s ease-out',
      mid: 'all 2s ease-out',
      slow: 'all 4s ease-out',
    },
    background: {
      fast: 'background-color 0.5s ease-out',
      mid: 'background-color 2s ease-out',
      slow: 'background-color 4s ease-out',
    },
    fontColor: {
      fast: 'color 0.5s ease-out',
      mid: 'color 2s ease-out',
      slow: 'color 4s ease-out',
    }
  }
};

export default theme;
