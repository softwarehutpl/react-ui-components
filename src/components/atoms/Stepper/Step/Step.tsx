import React from 'react';
import styled from 'styled-components';
import stepImage from '../../../../assets/register.svg';

export interface WrapperProps {
  width?: string;
  height?: string;
  color?: any;
  isActive?: boolean;
};

export interface IconProps {
  isActive?: boolean;
  color?: any;
};

export interface TitleProps {
  isActive?: boolean;
};

export interface StepProps extends WrapperProps, IconProps, TitleProps {
  stepClassName?: string;
  imageClassName?: string;
  image? : any;
  title?: string;
};

export const Wrapper = styled.div<WrapperProps>`
  position: relative; 
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: ${({ width }) => width};
  height: ${({ height }) => height};

  &:after {
    content: " ";
    position: absolute; 
    left: 50%;
    top: calc(50% + 35px);
    width: 100%;
    height: 3px;
    background-color: ${({ color, theme }) => (theme.colors[color].base)};
    z-index: -1;
  }
  &:last-child:after {
    display: none;
  }
`;

export const Image = styled.img`
  width: 112px;
  height: 90px;
  fill: red;
`;

export const Icon = styled.div<IconProps>`
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 70px;
  height: 70px;
  border-radius: 50%;
  border-color: ${( { theme, color }) => (theme.colors[color].base)};
  border-width: 3px;
  border-style: solid;
  background-color: ${({ color, theme, isActive }) => (isActive ? theme.colors[color].base : '#fff')};
    &:after {
      content: '';
      position: absolute;
      top: 9px;
      color: #fff;
      width: 20px;
      height: 30px;
      border: solid  #fff;
      border-width: 0 2px 2px 0;
      transform: rotate(45deg);
    }
`;

export const Title = styled.span<TitleProps>`
  font-weight: ${({ isActive }) =>  (isActive  && 500)};
`;

const Step: React.StatelessComponent<StepProps> = ({ 
  stepClassName,
  imageClassName,
  image, 
  title,
  isActive,
  color,
  width,
  height,
}) => (
  <Wrapper
    className={stepClassName}
    color={color}
    width={width}
    height={height}
    isActive={isActive}
  >
    <Image
      src={image ? image : stepImage}
      alt="step-image"
      className={imageClassName}
    />
    <Icon
      color={color}
      isActive={isActive}
    />
    <Title 
      isActive={isActive}
    >
      {title}
    </Title>
  </Wrapper>
);

Step.defaultProps = {
  stepClassName: '',
  color: 'primary',
  width: '100%',
  height: '300px',
  isActive: true,
};

export default Step;
