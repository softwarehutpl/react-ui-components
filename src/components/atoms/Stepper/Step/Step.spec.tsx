import React from 'react';
import { create, act } from 'react-test-renderer';
import 'jest-styled-components';
import Step, { Wrapper, Icon, Title } from './Step';
import theme from '../../../../common/theme';

describe('<Step />', () => {
  it('should render corectly', () => {
    const component = create(<Wrapper theme={theme} color="primary" />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render input with custom class', () => {
    const component = create(
      <Wrapper className="wrapper" theme={theme} color="secondary" />).toJSON();

    expect(component.props.className).toEqual(expect.stringContaining('wrapper'));
  });

  it('should render proper styling when isActve is false', () => {
    const icon= create( <Icon isActive={false} color="secondary" theme={theme} />).toJSON();
     expect(icon).toHaveStyleRule('background-color', '#fff');
  });

  it('should render proper styling when isActve is true and color is secondary', () => {
    const icon = create(<Icon isActive color="secondary" theme={theme} /> ).toJSON();
    const title = create( <Title isActive />).toJSON();

    expect(icon).toHaveStyleRule('background-color', '#2196f3');
    expect(title).toHaveStyleRule('font-weight', '500');
  });

  it('should render wrapper with proper style', () => {
    const wrapper = create(
      <Wrapper 
        isActive 
        theme={theme} 
        color="secondary" 
        width="10px"
        height="10px"
      />
    ).toJSON();

    expect(wrapper).toHaveStyleRule('width', '10px');
    expect(wrapper).toHaveStyleRule('height', '10px');
  });
});
