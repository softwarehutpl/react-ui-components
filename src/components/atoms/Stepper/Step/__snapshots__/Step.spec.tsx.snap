// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> should render corectly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:after {
  content: " ";
  position: absolute;
  left: 50%;
  top: calc(50% + 35px);
  width: 100%;
  height: 3px;
  background-color: #000000;
  z-index: -1;
}

.c0:last-child:after {
  display: none;
}

<div
  className="c0"
  color="primary"
/>
`;
